### PR TITLE
Update ticket.php

### PR DESCRIPTION
**Pull request by aftechro**
Re-make of Related tickets (total open and closed) and contact card, showing previous ticket and status of it. Also, made the contact mobile and contact phone clickable (tel:$contact_phone). made contact clickable on contact card

image

image

### CODE REVIEW

**Comment by wrongecho** [comment]
Test these changes at: https://patch11679.pr-review.itflow.org  
 _(automatic message)_

**Comment by wrongecho** [comment]
This looks great!

Small tweak: Can we please make it so it hides the previous ticket section if there isn't one?

**Comment by aftechro** [comment]
I`ll have to check that, but sure, i`ll give it a try :D

**Comment by aftechro** [comment]
Done now :D

File: ticket.php [hunk 1]
```diff
@@ -485,14 +502,51 @@
...
+
+  							  	$sql_prev_ticket = "SELECT ticket_id, ticket_created_at, ticket_subject, ticket_status, ticket_assigned_to FROM tickets WHERE ticket_contact_id = $contact_id AND ticket_id < $ticket_id ORDER BY ticket_id DESC LIMIT 1";
+    							$row = mysqli_fetch_assoc(mysqli_query($mysqli, $sql_prev_ticket));
+
+   								 if ($row) {
+        							$prev_ticket_id = $row['ticket_id'];
+    								//    $prev_ticket_created_at = $row['ticket_created_at'];
+        							$prev_ticket_subject = $row['ticket_subject'];
+        							$prev_ticket_status = $row['ticket_status'];
+    								//    $prev_ticket_assigned_to = $row['ticket_assigned_to'];
```
Comment (wrongecho) [inline ticket.php:528]: Could you please use intval (for the ID) and htmlentities (for the subject/status) when outputting onto the page? This helps prevent XSS attacks.

Like: $prev_ticket_subject = nullable_htmlentities($row['ticket_subject']);

**Comment by wrongecho** [comment]
Looking great dude! I really like what you've done with this. It's the info you need but not in your way and taking up loads of space. :)
Once the XSS is fixed I think this can be merged!

**Comment by aftechro** [comment]
that`s done now, hope i made it right :d 

only that code is a bit messier as i dont have that phpstorm license anymore to auto arrange it

**Comment by Heart1010** [comment]
A click at that "Open Tickets: 2" brings us now to the URL: ```.../tickets.php?contact_id=2&status=Open``` and not to ```.../client_tickets.php?client_id=2``` - is this correct 🤔 ?

**Comment by wrongecho** [comment]
> A click at that "Open Tickets: 2" brings us now to the URL: `.../tickets.php?contact_id=2&status=Open` and not to `.../client_tickets.php?client_id=2` - is this correct 🤔 ?

Hey! 
I agree it would be nicer if it took you to the client-specific view for tickets, however `client_tickets.php` doesn't have as much filtering built into it (yet) as `tickets.php` has so this currently isn't possible. 
This is definitely a great improvement idea though! Shouldn't be too much work, just need to copy across the logic from `tickets.php`.

**Comment by Heart1010** [comment]
```tickets.php?contact_id=2&status=Open``` here does **not** only show me the tickets from client with id 2 - it also shows me all the tickets from client 1 for example (I only have two clients store in my test installation).

**Comment by wrongecho** [comment]
Oh, I see what you mean - sorry! I admit I only tested this with a single client in the review.. The idea of this function (in my mind) was to just show previous tickets for that contact.

@aftechro - Seems like you forgot to include the code in `tickets.php` to actually allow filtering by contact_id?
image

**Comment by wrongecho** [comment]
@Heart1010 - I've [reverted](https://github.com/itflow-org/itflow/pull/685) the PR for now to prevent confusion. Good catch :)

**Comment by aftechro** [comment]
great, i'll have a look. i did not test it with multiple contacts and i missed that client_id&contact_id

**Pull request merged**